src/charm.py

```diff
@@ -1009,6 +1117,32 @@ def _get_tracing_endpoint_info(self) -> str:
     def _promtail_error(self, event: PromtailDigestError) -> None:
         logger.error(event.message)
 
+    def _configure_ingress(self, event: HookEvent) -> None:
+        """Method setting up the internal networking.
+
+        Since :class:`TraefikRouteRequirer` may not have been constructed with an existing
+        relation if a :class:`RelationJoinedEvent` comes through during the charm lifecycle, if we
+        get one here, we should recreate it, but OF will give us grief about "two objects claiming
+        to be ...", so manipulate its private `_relation` variable instead.
+
+        Args:
+            event: a :class:`HookEvent` to signal a change we may need to respond to.
+        """
+        if not self.unit.is_leader():
+            return
+
+        # If it's a RelationJoinedEvent, set it in the ingress object
+        if isinstance(event, RelationJoinedEvent):
+            self.internal_ingress._relation = event.relation
```

Comment:
Why is this needed?

Reply:
not sure to be fair, implementation is matching grafana's one

happy to remove and try to see if it works

Reply:
chatted and all hell breaks loose if removed